File: Bug_Hunt_Practice/Question 6/solution.c
#include <stdio.h>

void printArray(int arr[], int size) {
    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

void reverseArray(int arr[], int size) {
    for (int i = 0; i < size; i++) {
        int temp = arr[i];
        arr[i] = arr[size - 1 - i];
        arr[size - 1 - i] = temp;
    }
}

int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int size = 5;
    
    reverseArray(arr, size);
    
    printf("Expected Output: 5 4 3 2 1\n");
    printf("Actual Output:   ");
    printArray(arr, size);
    return 0;
}

File: Bug_Hunt_Practice/Question 6/solution.py
def reverse_array(arr):
    size = len(arr)
    
    for i in range(size):
        temp = arr[i]
        arr[i] = arr[size - 1 - i]
        arr[size - 1 - i] = temp


# --- Main execution ---
my_list = [1, 2, 3, 4, 5]
reverse_array(my_list)

print(f"Expected Output: [5, 4, 3, 2, 1]")
print(f"Actual Output:   {my_list}")